cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -90,90 +103,102 @@ renderShelleyQueryCmdError err =
 runQueryCmd :: QueryCmd -> ExceptT ShelleyQueryCmdError IO ()
 runQueryCmd cmd =
   case cmd of
-    QueryProtocolParameters' consensusModeParams network mOutFile ->
-      runQueryProtocolParameters consensusModeParams network mOutFile
-    QueryTip consensusModeParams network mOutFile ->
-      runQueryTip consensusModeParams network mOutFile
-    QueryStakeDistribution' consensusModeParams network mOutFile ->
-      runQueryStakeDistribution consensusModeParams network mOutFile
-    QueryStakeAddressInfo consensusModeParams addr network mOutFile ->
-      runQueryStakeAddressInfo consensusModeParams addr network mOutFile
-    QueryLedgerState' consensusModeParams network mOutFile ->
-      runQueryLedgerState consensusModeParams network mOutFile
-    QueryProtocolState' consensusModeParams network mOutFile ->
-      runQueryProtocolState consensusModeParams network mOutFile
-    QueryUTxO' consensusModeParams qFilter networkId mOutFile ->
-      runQueryUTxO consensusModeParams qFilter networkId mOutFile
+    QueryProtocolParameters' era consensusModeParams network mOutFile ->
+      runQueryProtocolParameters era consensusModeParams network mOutFile
+    QueryTip era consensusModeParams network mOutFile ->
+      runQueryTip era consensusModeParams network mOutFile
+    QueryStakeDistribution' era consensusModeParams network mOutFile ->
+      runQueryStakeDistribution era consensusModeParams network mOutFile
+    QueryStakeAddressInfo era consensusModeParams addr network mOutFile ->
+      runQueryStakeAddressInfo era consensusModeParams addr network mOutFile
+    QueryLedgerState' era consensusModeParams network mOutFile ->
+      runQueryLedgerState era consensusModeParams network mOutFile
+    QueryStakeSnapshot' era consensusModeParams network poolid ->
+      runQueryStakeSnapshot era consensusModeParams network poolid
+    QueryPoolParams' era consensusModeParams network poolid ->
+      runQueryPoolParams era consensusModeParams network poolid
+    QueryProtocolState' era consensusModeParams network mOutFile ->
+      runQueryProtocolState era consensusModeParams network mOutFile
+    QueryUTxO' era consensusModeParams qFilter networkId mOutFile ->
+      runQueryUTxO era consensusModeParams qFilter networkId mOutFile
 
 runQueryProtocolParameters
-  :: AnyConsensusModeParams
+  :: AnyCardanoEra
+  -> AnyConsensusModeParams
   -> NetworkId
   -> Maybe OutputFile
   -> ExceptT ShelleyQueryCmdError IO ()
-runQueryProtocolParameters (AnyConsensusModeParams cModeParams) network mOutFile = do
+runQueryProtocolParameters anyEra@(AnyCardanoEra era) (AnyConsensusModeParams cModeParams)
+                           network mOutFile = do
   SocketPath sockPath <- firstExceptT ShelleyQueryCmdEnvVarSocketErr
                            readEnvSocketPath
+
+  let consensusMode = consensusModeOnly cModeParams
```

Comment:
I'm not sure what's happened here. If you look at the diff we have clearly undone some refactoring.

Reply:
I think this was done by the git merge.  I've remerged on the latest sources